src/app/home/pages/ClustersPage/components/AddEditClusterModal/__tests__/AddEditClusterModal.test.tsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom';
+
+import { createStore } from 'redux';
+import { Provider } from 'react-redux';
+import rootReducer from '../../../../../../../reducers';
+
+import AddEditClusterModal from '../index';
+
+const store = createStore(rootReducer, {});
+
+describe('<AddEditClusterModal />', () => {
+  it('adds a cluster with valid values', () => {
+    const initialProps = {
+      isOpen: true,
+    };
+
+    render(
+      <Provider store={store}>
+        <AddEditClusterModal {...initialProps} />
+      </Provider>
+    );
+
+    userEvent.type(screen.getByTestId('cluster-name'), 'clustername');
+    userEvent.type(screen.getByTestId('url'), 'http://example.com');
+    userEvent.type(screen.getByTestId('token'), 'qwertyuiop');
+
+    expect(screen.getByDisplayValue('clustername')).toBeInTheDocument;
+    expect(screen.getByDisplayValue('http://example.com')).toBeInTheDocument;
+    expect(screen.getByDisplayValue('qwertyuiop')).toBeInTheDocument;
+    expect(screen.getByRole('button', { name: 'Add cluster' })).toBeEnabled;
+  });
+
+  it('cannot add a cluster with unvalid cluster name', () => {
+    const initialProps = {
+      isOpen: true,
+    };
+
+    render(
+      <Provider store={store}>
+        <AddEditClusterModal {...initialProps} />
+      </Provider>
+    );
+
+    userEvent.type(screen.getByTestId('cluster-name'), '-clustername');
+    userEvent.type(screen.getByTestId('url'), 'example.com');
+
+    expect(screen.getByRole('button', { name: 'Add cluster' })).toBeDisabled;
+  });
+
+  it('cannot add a cluster with unvalid cluster url', () => {
+    const initialProps = {
+      isOpen: true,
+    };
+
+    render(
+      <Provider store={store}>
+        <AddEditClusterModal {...initialProps} />
+      </Provider>
+    );
+
+    userEvent.type(screen.getByTestId('url'), 'example.com');
+
+    expect(screen.getByRole('button', { name: 'Add cluster' })).toBeDisabled;
+  });
+
+  it('updates a cluster with valid fields', () => {
```

Comment:
same here, maybe "properly fills initialClusterValues and enables add cluster button when valid" or something. Not necessarily that, just thinking ahead in case we want to add more integration-style tests that involve redux actions to actually add/update the cluster.